### PR TITLE
8282072: G1: Rename CardSetPtr to CardSetContainerPtr

### PR DESCRIPTION
**Pull request by walulyai**
Hi all,
Please review this change to rename CardSetPtr to ContainerPtr since it is a pointer to the containers. 

Testing: tier1

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8282072](https://bugs.openjdk.java.net/browse/JDK-8282072): G1: Rename CardSetPtr to CardSetContainerPtr

### Reviewers
 * [Albert Mingkun Yang](https://openjdk.java.net/census#ayang) (@albertnetymk - **Reviewer**)
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7816/head:pull/7816` \
`$ git checkout pull/7816`

Update a local copy of the PR: \
`$ git checkout pull/7816` \
`$ git pull https://git.openjdk.java.net/jdk pull/7816/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7816`

View PR using the GUI difftool: \
`$ git pr show -t 7816`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7816.diff">https://git.openjdk.java.net/jdk/pull/7816.diff</a>

</details>

### CODE REVIEW

**Review by albertnetymk — APPROVED** [review]

File: src/hotspot/share/gc/g1/g1CardSet.hpp [hunk 1]
```diff
@@ -221,19 +218,19 @@ class G1CardSet : public CHeapObj<mtGCCardSet> {
   static const uintptr_t CardSetHowl           = 0x3;
 
   // The special sentinel values
-  static constexpr CardSetPtr FreeCardSet = nullptr;
-  // Unfortunately we can't make (G1CardSet::CardSetPtr)-1 constexpr because
+  static constexpr ContainerPtr FreeCardSet    = nullptr;
```
Comment (tschatzl) [inline src/hotspot/share/gc/g1/g1CardSet.hpp:221]:
```suggestion
  static constexpr ContainerPtr FreeCardSet = nullptr;
```
Let's avoid lining up fairly unrelated constants.

File: src/hotspot/share/gc/g1/g1CardSet.hpp [hunk 2]
```diff
@@ -221,19 +218,19 @@ class G1CardSet : public CHeapObj<mtGCCardSet> {
...
+  // Unfortunately we can't make (G1CardSet::ContainerPtr)-1 constexpr because
   // reinterpret_casts are forbidden in constexprs. Use a regular static instead.
-  static CardSetPtr FullCardSet;
+  static ContainerPtr FullCardSet;
 
-  static const uintptr_t CardSetPtrTypeMask    = ((uintptr_t)1 << CardSetPtrHeaderSize) - 1;
+  static const uintptr_t ContainerPtrTypeMask  = ((uintptr_t)1 << ContainerPtrHeaderSize) - 1;
```
Comment (tschatzl) [inline src/hotspot/share/gc/g1/g1CardSet.hpp:226]:
```suggestion
  static const uintptr_t ContainerPtrTypeMask = ((uintptr_t)1 << ContainerPtrHeaderSize) - 1;
```
Same here.

File: src/hotspot/share/gc/g1/g1CardSet.hpp [hunk 3]
```diff
@@ -245,42 +242,42 @@ class G1CardSet : public CHeapObj<mtGCCardSet> {
...
+  void release_and_must_free_container(ContainerPtr container);
+
+  // Coarsens the card set container cur_container to the next level; tries to replace the
+  // previous ContainerPtr with a new one which includes the given card_in_region.
+  // coarsen_container does not transfer cards from cur_container
+  // to the new container. Transfer is achieved by transfer_cards.
+  // Returns true if this was the thread that coarsened the container (and added the card).
+  bool coarsen_container(ContainerPtr volatile* container_addr,
+                        ContainerPtr cur_container,
                         uint card_in_region, bool within_howl = false);
```
Comment (tschatzl) [inline src/hotspot/share/gc/g1/g1CardSet.hpp:262]:
```suggestion
  bool coarsen_container(ContainerPtr volatile* container_addr,
                         ContainerPtr cur_container,
                         uint card_in_region, bool within_howl = false);
```

File: src/hotspot/share/gc/g1/g1CardSet.hpp [hunk 4]
```diff
@@ -302,8 +299,8 @@ class G1CardSet : public CHeapObj<mtGCCardSet> {
   G1CardSet(G1CardSetConfiguration* config, G1CardSetMemoryManager* mm);
   virtual ~G1CardSet();
 
-  // Adds the given card to this set, returning an appropriate result. If added,
-  // updates the total count.
+  // Adds the given card to this set, returning an appropriate result.
+  // If added, updates the total count.
```
Comment (tschatzl) [inline src/hotspot/share/gc/g1/g1CardSet.hpp:303]:
```suggestion
  // If incremental_count is true and the card has been added, updates the total count.
```

File: src/hotspot/share/gc/g1/g1CardSet.hpp [hunk 5]
```diff
@@ -194,23 +191,23 @@ class G1CardSet : public CHeapObj<mtGCCardSet> {
...
+  // 1...11111 full               All card indexes in the whole area this ContainerPtr covers are part of this container.
+  // X...XXX00 inline-ptr-cards   A handful of card indexes covered by this ContainerPtr are encoded within the ContainerPtr.
   // X...XXX01 array of cards     The container is a contiguous array of card indexes.
   // X...XXX10 bitmap             The container uses a bitmap to determine whether a given index is part of this set.
-  // X...XXX11 howl               This is a card set container containing an array of CardSetPtr, with each CardSetPtr
+  // X...XXX11 howl               This is a card set container containing an array of ContainerPtr, with each ContainerPtr
   //                              limited to a sub-range of the original range. Currently only one level of this
   //                              container is supported.
-  typedef void* CardSetPtr;
+  using ContainerPtr = void*;
```
Comment (tschatzl) [inline src/hotspot/share/gc/g1/g1CardSet.hpp:210]: I can't anchor this elsewhere: the below `CardSetInlinePtr` etc tag names should also be `ContainerInlinePtr` etc, shouldn't they?

**Review by tschatzl — APPROVED** [review]
Lgtm with some minor formatting comments.

**Review by tschatzl — APPROVED** [review]

**Review by albertnetymk — APPROVED** [review]

**Comment by walulyai** [comment]
Thanks @albertnetymk and @tschatzl for the reviews.

/integrate

**Pull request closed without merging**